### ISSUE TITLE
[v.2.6.0] Release Tracker

### ISSUE THREAD

**Issue by kit1980** [pytorch/pytorch#142814]
We cut a [release branch](https://github.com/pytorch/pytorch/tree/release/2.6) for the 2.6.0 release.

Our plan from this point from this point is roughly:

* Phase 1 (until 1/13/25): work on finalizing the release branch
* Phase 2 (after 1/13/25): perform extended integration/stability/performance testing based on Release Candidate builds.

This issue is for tracking cherry-picks to the release branch.

## Cherry-Pick Criteria

**Phase 1 (until 1/13/25):**

Only low-risk changes may be cherry-picked from main:

1. Fixes to regressions against the most recent minor release (e.g. 2.5.x for this release; see [module: regression issue list](https://github.com/pytorch/pytorch/issues?q=is%3Aissue+is%3Aopen+label%3A%22module%3A+regression%22+))
2. Critical fixes for: [silent correctness](https://github.com/pytorch/pytorch/issues?q=is%3Aissue+is%3Aopen+label%3A%22topic%3A+correctness+%28silent%29%22), [backwards compatibility](https://github.com/pytorch/pytorch/issues?q=is%3Aissue+is%3Aopen+label%3A%22topic%3A+bc-breaking%22+), [crashes](https://github.com/pytorch/pytorch/issues?q=is%3Aissue+is%3Aopen+label%3A%22topic%3A+crash%22+), [deadlocks](https://github.com/pytorch/pytorch/issues?q=is%3Aissue+is%3Aopen+label%3A%22topic%3A+deadlock%22+), (large) [memory leaks](https://github.com/pytorch/pytorch/issues?q=is%3Aissue+is%3Aopen+label%3A%22topic%3A+memory+usage%22+)
3. Critical fixes to new features introduced in the most recent minor release (e.g. 2.5.x for this release)
4. Test/CI fixes
5. Documentation improvements
6. Compilation fixes or ifdefs required for different versions of the compilers or third-party libraries
7. Release branch specific changes (e.g. change version identifiers)

Any other change requires special dispensation from the release managers (currently @atalman, @malfet, @kit1980). If this applies to your change please write "Special Dispensation" in the "Criteria Category:" template below and explain.

**Phase 2 (after 1/13/25):**

Note that changes here require us to rebuild a Release Candidate and restart extended testing (likely delaying the release). Therefore, the only accepted changes are **Release-blocking** critical fixes for: [silent correctness](https://github.com/pytorch/pytorch/issues?q=is%3Aissue+is%3Aopen+label%3A%22topic%3A+correctness+%28silent%29%22), [backwards compatibility](https://github.com/pytorch/pytorch/issues?q=is%3Aissue+is%3Aopen+label%3A%22topic%3A+bc-breaking%22+), [crashes](https://github.com/pytorch/pytorch/issues?q=is%3Aissue+is%3Aopen+label%3A%22topic%3A+crash%22+), [deadlocks](https://github.com/pytorch/pytorch/issues?q=is%3Aissue+is%3Aopen+label%3A%22topic%3A+deadlock%22+), (large) [memory leaks](https://github.com/pytorch/pytorch/issues?q=is%3Aissue+is%3Aopen+label%3A%22topic%3A+memory+usage%22+)

Changes will likely require a discussion with the larger release team over VC or Slack.

## Cherry-Pick Process

1. Ensure your PR has landed in master. This does not apply for release-branch specific changes (see Phase 1 criteria).
2. Create (but do not land) a PR against the [release branch](https://github.com/pytorch/pytorch/tree/release/2.6).
   <details>

    ```bash
    # Find the hash of the commit you want to cherry pick
    # (for example, abcdef12345)
    git log

    git fetch origin release/2.6
    git checkout release/2.6
    git cherry-pick -x abcdef12345

    # Submit a PR based against 'release/2.6' either:
    # via the GitHub UI
    git push my-fork

    # via the GitHub CLI
    gh pr create --base release/2.6
    ```
 
    You can also use the `@pytorchbot cherry-pick` command to cherry-pick your PR. To do this, just add a comment in your merged PR. For example:
   
    ```
    @pytorchbot cherry-pick --onto release/2.6 -c docs
    ```
    (`-c docs` - is the category of your changes - adjust accordingly):
    
    For more information, see [pytorchbot cherry-pick docs](https://github.com/pytorch/pytorch/wiki/Bot-commands#cherry-pick).

    </details>

  

3. Make a request below with the following format:

```
Link to landed trunk PR (if applicable):
* 

Link to release branch PR:
* 

Criteria Category:
* 
```

1. Someone from the release team will reply with approved / denied or ask for more information.
2. If approved, someone from the release team will merge your PR once the tests pass. **Do not land the release branch PR yourself.**

**NOTE: Our normal tools (ghstack / ghimport, etc.) do not work on the release branch.**

Please note HUD Link with branch CI status and link to the HUD to be provided here.
[HUD](https://hud.pytorch.org/hud/pytorch/pytorch/release%2F2.6)

### Versions

2.6.0

cc @seemethere @malfet @pytorch/pytorch-dev-infra

**Comment by atalman** [comment]
Link to landed trunk PR (if applicable):
* NA

Link to release branch PR:
* https://github.com/pytorch/pytorch/pull/142864

Criteria Category:
* Release Only

**Comment by kit1980** [comment]
Closing this temporarily until new branch cut.

**Comment by kit1980** [comment]
Link to landed trunk PR (if applicable):
* 

Link to release branch PR:
* https://github.com/pytorch/pytorch/pull/143092

Criteria Category:
Critical - reverts

**Comment by atalman** [comment]
Link to landed trunk PR (if applicable):
* NA

Link to release branch PR:
* https://github.com/pytorch/pytorch/pull/143085/

Criteria Category:
* release only

**Comment by kit1980** [comment]
Link to landed trunk PR (if applicable):
* https://github.com/pytorch/pytorch/pull/143125

Link to release branch PR:
* https://github.com/pytorch/pytorch/pull/143131

Criteria Category:
Critical

**Comment by XuehaiPan** [comment]
Link to landed trunk PR (if applicable):
* #143186

Link to release branch PR:
* #140542

Criteria Category:
* Documentation improvements

**Comment by jiayisunx** [comment]
Link to landed trunk PR (if applicable):
* https://github.com/pytorch/pytorch/pull/143279

Link to release branch PR:
* https://github.com/pytorch/pytorch/pull/143452

Criteria Category: Critical - Inductor bug fix

**Comment by XuehaiPan** [comment]
Link to landed trunk PR (if applicable):
* #137398
* #137399

Link to release branch PR:
* #143499

Criteria Category:
* Compilation fixes for third-party libraries

**Comment by williamwen42** [comment]
Link to landed trunk PR (if applicable):
* https://github.com/pytorch/pytorch/pull/143404

Link to release branch PR:
* https://github.com/pytorch/pytorch/pull/143594

Criteria Category:
* Fix new feature: raise error if torch.compile is used in 3.13t (nogil), since it is not supported.

**Comment by jiayisunx** [comment]
Link to landed trunk PR (if applicable):
* https://github.com/pytorch/pytorch/pull/141759

Link to release branch PR:
* https://github.com/pytorch/pytorch/pull/143617

Criteria Category: Critical - Inductor performance regression ( it's a regression compared to 2.5.1).

**Comment by guangyey** [comment]
Link to landed trunk PR (if applicable):
* https://github.com/pytorch/pytorch/pull/143541

Link to release branch PR:
* https://github.com/pytorch/pytorch/pull/143781

Criteria Category: Critical - New API torch.accelerator.xxx (introduced in 2.6) bug fix

---

**Comment by ngimel** [comment]
Link to landed trunk PR (if applicable):
* #143682 

Link to release branch PR:
* #143875

Criteria Category:
* Critical fix for silent correcness

**Comment by FurkanGozukara** [comment]
how can i install 2.6.0 and test?
this fails
`pip3 install torch==2.6.0 torchvision torchaudio --index-url https://download.pytorch.org/whl/cu124`

**Comment by kit1980** [comment]
> how can i install 2.6.0 and test? this fails `pip3 install torch==2.6.0 torchvision torchaudio --index-url https://download.pytorch.org/whl/cu124`

@FurkanGozukara 
pip3 install torch==2.6.0 --index-url https://download.pytorch.org/whl/test/cu124

**Comment by malfet** [comment]
Link to landed trunk PR:
 - https://github.com/pytorch/pytorch/pull/143934
 - https://github.com/pytorch/pytorch/pull/144055

Link to release branch PR:
-  https://github.com/pytorch/pytorch/pull/144096

Criteria Category: Regression against previous minor release as well as in newly introduced feature

**Comment by EikanWang** [comment]
Link to landed trunk PR (if applicable):
* https://github.com/pytorch/pytorch/pull/143984

Link to release branch PR:
* https://github.com/pytorch/pytorch/pull/144209

Criteria Category:
* Fix End-2-End accuracy regression due to batch_norm accuracy issue and LSRM/RNN implementation issue.

**Comment by bhack** [comment]
Is there a way to regularly test daily 2.6.0 branch wheels before RC? 
As nightlies and nightlies images are now on 2.7.0.

**Comment by isuruf** [comment]
@atalman, the regression was introduced by https://github.com/pytorch/pytorch/pull/140597 where the precedence was changed. It was merged after 2.5 release. The fix was not to change https://github.com/pytorch/pytorch/blob/main/torch/utils/_sympy/functions.py#L407 back, but to fix `sympystr` method. Let me know if you need more info.

**Comment by stas00** [comment]
The OP seems to have a typo:

Phase 1 (until 1/13/25):
Phase 2 (until 1/13/25):

can't be the same date, no?

**Comment by kit1980** [comment]
Link to landed trunk PR (if applicable):
* #143682 

Link to release branch PR:
* https://github.com/pytorch/pytorch/pull/144735

Criteria Category:
* Critical fix for silent correctness
(manual cherry-pick after revert/re-merge https://github.com/pytorch/pytorch/issues/142814#issuecomment-2563028302)

**Comment by wanchaol** [comment]
Link to landed trunk PR (if applicable):
* https://github.com/pytorch/pytorch/pull/144099

Link to release branch PR:
* https://github.com/pytorch/pytorch/pull/144740

Criteria Category:
* documentation

**Comment by wanchaol** [comment]
Link to landed trunk PR (if applicable):
* https://github.com/pytorch/pytorch/pull/144100

Link to release branch PR:
* https://github.com/pytorch/pytorch/pull/144741

Criteria Category:
* documentation

**Comment by malfet** [comment]
Link to landed trunk PRs:
- https://github.com/pytorch/pytorch/pull/144716
- https://github.com/pytorch/pytorch/pull/144698

Link to cherry-pick PR:
- https://github.com/pytorch/pytorch/pull/144878

Criteria category: RelEng?

**Comment by kit1980** [comment]
No longer accepting cherry-pick requests after this point.

**Comment by kit1980** [comment]
No longer accepting cherry-pick requests after this point.

**Comment by atalman** [comment]
Link to landed trunk PR (if applicable):
* NA

Link to release branch PR:
* https://github.com/pytorch/pytorch/pull/145616

Criteria Category:
* Release Only changes, improve file size of CUDA 12.6 binaries

**Comment by atalman** [comment]
Please note: We are no longer accepting cherry-pick requests after this point.

**Comment by kit1980** [comment]
Please note: We are no longer accepting cherry-pick requests after this point.
